src/mpid/ch4/src/ch4_wait.h

```diff
@@ -45,8 +46,15 @@ MPL_STATIC_INLINE_PREFIX void MPIDI_set_progress_vci_n(int n, MPIR_Request ** re
         if (!found) {
             state->vci[idx++] = vci;
         }
+#ifndef MPIDI_CH4_DIRECT_NETMOD
+        local_only = local_only && MPIDI_REQUEST(reqs[i], is_local);
+#endif
     }
     state->vci_count = idx;
+
+    if (local_only) {
+        state->flag &= ~MPIDI_PROGRESS_NM;
```

Comment:
We should still perform netmod progress during "global" progress. 

https://github.com/pmodels/mpich/blob/f47908fa4a74bf4ac29997202fb2967c8c59b0c9/src/mpid/ch4/src/ch4_progress.h#L78
```
        if (state->flag & MPIDI_PROGRESS_NM && (is_global || !made_progress)) { \
```
Make it 
```
        if (is_global || (state->flag & MPIDI_PROGRESS_NM && !made_progress)) { \
```